lib/Data/Decimate.pm

```diff
@@ -73,11 +73,11 @@ sub decimate {
     my $decimate_epoch = do {
         use integer;
         (($el->{epoch} + $interval - 1) / $interval) * $interval;
-    } if $el;
+    } if $data->[0];
```

Comment:
Technically this is wrong, since you're combining `my` with a conditional, so the old value will be left over. However, since we only push when the condition is true, this shouldn't cause any problems. Will raise a separate PR to fix it.